explorer/client/src/pages/object-result/ObjectResult.tsx

```diff
@@ -47,53 +42,30 @@ const Fail = ({ objID }: { objID: string | undefined }): JSX.Element => {
     );
 };
 
-// Get the data for the object ID and either:
-// address that publishes a Package or
-// module and package associated with token
+// Get the data for the object ID and address that publishes a Package:
 function getObjectDataWithPackageAddress(objID: string, network: string) {
     return rpc(network)
         .getObject(objID as string)
         .then((objState) => {
             const resp: DataType = translate(objState) as DataType;
-            if (resp.data.tx_digest) {
+            if (resp.objType === 'Move Package' && resp.data.tx_digest) {
                 return rpc(network)
                     .getTransactionWithEffects(resp.data.tx_digest)
-                    .then((txEff) => {
-                        if (resp.objType === 'Move Package') {
-                            // If Package, then extract publisher address
-                            return {
-                                ...resp,
-                                publisherAddress: getTransactionSender(
-                                    txEff.certificate
-                                ),
-                            };
-                        } else {
-                            // If Token, then extract the module and package
-                            const movecall = getMoveCallTransaction(
-                                getTransactions(txEff.certificate)[0]
-                            );
-                            if (!movecall) return resp;
-                            return {
-                                ...resp,
-                                module: movecall.module,
-                                package: getObjectId(movecall.package),
-                            };
-                        }
-                    })
+                    .then((txEff) => ({
+                        ...resp,
+                        publisherAddress: getTransactionSender(
+                            txEff.certificate
+                        ),
+                    }))
                     .catch((err) => {
                         console.log(err);
                         // TODO: Not sure if I should show Genesis as Package Publisher or ignore it
-
-                        if (resp.objType === 'Move Package') {
-                            return {
-                                ...(resp.owner === 'Immutable'
-                                    ? { publisherAddress: 'Genesis' }
-                                    : {}),
-                                ...resp,
-                            };
-                        } else {
-                            return resp;
-                        }
+                        return {
+                            ...(resp.owner === 'Immutable'
+                                ? { publisherAddress: 'Genesis' }
+                                : {}),
```

Comment:
This logic is wrong and should not be in the catch block. Instead, on line 52 we should do 

```
const GENESIS_TX_DIGEST = 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=`;
if (resp.data.tx_digest === GENESIS_TX_DIGEST) {
	return { publisherAddress: 'Genesis' };
}
```